### PR TITLE
Provide AVRO plugin

### PR DESCRIPTION
**Pull request by dmsolr**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

## Provide Avro plugin
checklist:
- [x] update component-libraries.yml
- [x] update supported-library.md
- [x] add ico into skywalking-web
- [x] testcase
- [x] unit test

## Screenshot 
image

image

image

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/avro-plugin/src/main/java/org/apache/skywalking/apm/plugin/avro/AbstractRequestInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            Protocol protocol = (Protocol)allArguments[0];
+            Transceiver transceiver = (Transceiver)allArguments[1];
+            try {
+                objInst.setSkyWalkingDynamicField(new AvroInstance(
+                    protocol.getNamespace() + "." + protocol.getName() + ".",
+                    transceiver.getRemoteName()
+                ));
+            } catch (IOException e) {
+                objInst.setSkyWalkingDynamicField(new AvroInstance("Undefined", "Undefined"));
+                LOG.error("", e);
```
Comment (arugal) [inline apm-sniffer/apm-sdk-plugin/avro-plugin/src/main/java/org/apache/skywalking/apm/plugin/avro/AbstractRequestInterceptor.java]: Why empty string.

**Comment by dmsolr** [comment]
> One question, In `Avro` document, it is
> 
> 1. Apache Avro™ is a data serialization system.
> 2. Remote procedure call (RPC).
> 
> How does this PR works, if Avro is only working as a serialization tool only?

Yes, we support RPC only now. Don't have a plan to support the part of data serialization.

**Comment by wu-sheng** [comment]
Please fix the icon related PR.

File: apm-sniffer/apm-sdk-plugin/avro-plugin/src/main/java/org/apache/skywalking/apm/plugin/avro/SWClientRPCPlugin.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.apache.skywalking.apm.agent.core.context.ContextCarrier;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+
+import java.nio.ByteBuffer;
+
+public class SWClientRPCPlugin extends RPCPlugin {
+
+    @Override
+    public void clientSendRequest(RPCContext context) {
+        if (ContextManager.isActive()) {
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/avro-plugin/src/main/java/org/apache/skywalking/apm/plugin/avro/SWClientRPCPlugin.java]: Is this check really meaningful? Such as in some cases, the context wouldn't be active? This Check is also a payload.

File: test/plugin/scenarios/avro-scenario/config/expectedData.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+              - {parentEndpointId: 0, parentEndpoint: /avro-scenario/case/avro-scenario, networkAddressId: 0,
+                 entryEndpointId: 0, refType: CrossProcess, parentSpanId: 1, parentTraceSegmentId: not null,
+                 parentServiceInstanceId: 2, networkAddress: 'localhost/127.0.0.1:9018', entryEndpoint: /avro-scenario/case/avro-scenario,
+                 entryServiceInstanceId: 2}
+  - applicationCode: avro-client
+    segmentSize: ge 2
+    segments:
+      - segmentId: not null
+        spans:
+          - {operationName: example.proto.Greeter.hello, operationId: 0, parentSpanId: 0,
```
Comment (wu-sheng) [inline test/plugin/scenarios/avro-scenario/config/expectedData.yaml]: Don't use this kind of format. Please change it to normal one.

**Comment by wu-sheng** [comment]
Any issue or just unstable CI?

**Comment by wu-sheng** [comment]
@dmsolr Avro test failing?

**Pull request merged**